mathics/builtin/combinatorial.py

```diff
@@ -266,3 +266,923 @@ class RogersTanimotoDissimilarity(_BooleanDissimilarity):
+            'DagumDistribution[p, a, b]',
+            'ExpGammaDistribution[k, theta, mu]',
+            'TriangularDistribution[{min, max}, c]',
+            'CauchyDistribution[a, b]',
+            'BinormalDistribution[{sigma1, sigma2}, rho]',
+            'LogSeriesDistribution[theta]',
+            'ZipfDistribution[n, rho]',
+            'KDistribution[nu, w]',
+            'BetaPrimeDistribution[p, q, alpha, beta]',
+            'TukeyLambdaDistribution[lambda]',
+            'SinghMaddalaDistribution[q, a, b]',
+            'RayleighDistribution[sigma]',
+            'InverseChiSquareDistribution[nu]',
+            'WaringYuleDistribution[alpha, beta]',
+            'TukeyLambdaDistribution[lambda, mu, sigma]',
+            'GeometricDistribution[p]',
+            'NoncentralBetaDistribution[alpha, beta, delta]',
+            'NormalDistribution[mu, sigma]',
+            'WakebyDistribution[alpha, beta, gamma, delta, mu]']
+
+
+def Quantile(Builtin):
+    """
+    <dl>
+    <dt>'Quantile[$list$, $q$]'
+        <dd>gives the qth quantile of list'.      
+    </dl>
+
+    >> Quantile[{1, 2, 3, 4, 5, 6, 7}, 1/2]
+     = 4
+    """
+    
+    ##TODO: implement other patterns
+    
+    rules = {'Quantile[list_, q_]':'Sort[list, Less][[Ceiling[q*Length[list]]]]'}
+    
+
+
+class Distribution(Builtin):
+    ##todo
+        
+    def cdf_sym(self, symbol):
+
+
+
+    def expectation(self):
+
+
+
+    def probability(self, conditions):
+
+
+
+class DistributionParameterQ(Builtin):
+
+    def apply(self, expr, evaluation):
+        'DistributionParameterQ[expr]'
+    ##todo
+
+
+class DistributionParameterAssumptions(Builtin):
+
+    def apply(self, distribution, evaluation):
+        'DistributionParameterAssumptions[distribution]'
+
+        if isinstance(distribution, Distribution):
+            return String(distribution.arg_assumptions)
+
+
+class CDF(Builtin):
+
+    def apply(self, distribution, evaluation):
+        'CDF[distribution, symbol]'
+
+        if isinstance(distribution, Distribution):
+            return distribution.cdf(str(symbol)))
+            
+
+class PDF(Builtin):
+
+    def apply(self, distribution, evaluation):
+        'PDF[distribution, symbol]'
+
+        if isinstance(distribution, Distribution):
+            return distribution.pdf(str(symbol)))
+
+
+class InverseCDF(Builtin):
+
+
+
+class SurvivalFunction(Builtin):
+
+    def apply(self, distribution, evaluation):
+        'SurvivalFunction[distribution]'
+        
+        return Expression(evaluate('1 - CDF[distribution, x]'))
+
+    
+class InverseSurvivalFunction(Builtin):
+
+
+    
+class SurvivalDistribution(Builtin):
+
+
+    
+class BernoulliDistribution(Builtin):
+
+
+
+class BetaBinomialDistribution(Distribution):
+
+
+
+class BetaDistribution(Distribution):
+
+    'BetaDistribution[alpha, beta]'
+    
+    parameters = [alpha, beta]
+    nargs = 2
+    sympy_name = 'Beta'
+    arg_assumptions = 'alpha > 0 && beta > 0'
+
```

Comment:
I plan to call the sympy corresponding function (parameters are nearly the same)